backends/rapidpro/backend_test.go

```diff
@@ -929,6 +930,71 @@ func (ts *BackendTestSuite) TestOutgoingQueue() {
 	ts.False(sent)
 }
 
+func (ts *BackendTestSuite) TestOutgoingStatusUpdateOutOfOrder() {
+	// add one of our outgoing messages to the queue
+	ctx := context.Background()
+	r := ts.b.rp.Get()
+	defer r.Close()
+
+	dbMsg := readMsgFromDB(ts.b, 10000)
+	dbMsg.ChannelUUID_ = courier.ChannelUUID("dbc126ed-66bc-4e28-b67b-81dc3327c95d")
+	ts.NotNil(dbMsg)
+
+	// serialize our message
+	msgJSON, err := json.Marshal([]any{dbMsg})
+	ts.NoError(err)
+
+	err = queue.PushOntoQueue(r, msgQueueName, "dbc126ed-66bc-4e28-b67b-81dc3327c95d", 10, string(msgJSON), queue.HighPriority)
+	ts.NoError(err)
+
+	// pop a message off our queue
+	msg, err := ts.b.PopNextOutgoingMsg(ctx)
+	ts.NoError(err)
+	ts.NotNil(msg)
+
+	clog := courier.NewChannelLog(courier.ChannelLogTypeUnknown, msg.Channel(), nil)
+
+	// make sure it is the message we just added
+	ts.Equal(dbMsg.ID(), msg.ID())
+
+	// and that it has the appropriate text
+	ts.Equal(msg.Text(), "test message")
+
+	// this message should not be marked as sent
+	sent, err := ts.b.WasMsgSent(ctx, msg.ID())
+	ts.NoError(err)
+	ts.False(sent)
+
+	// write an failed status
+	err = ts.b.WriteStatusUpdate(ctx, ts.b.NewStatusUpdate(msg.Channel(), msg.ID(), courier.MsgStatusFailed, clog))
+	time.Sleep(time.Second) // give committer time to write this
+	ts.NoError(err)
+
+	m := readMsgFromDB(ts.b, msg.ID())
+	ts.Equal(m.Status_, courier.MsgStatusFailed)
```

Comment:
@rowanseymour any idea why this line fails for package tests but pass for the individual test?

Reply:
I found that https://github.com/nyaruka/courier/pull/796/files#diff-3129da45587845bddfafeaa91c8bf98aec96cd9dd1d728b0ad845463ae7735bfL570